File: Individual Assingment/TASK1/code/c++/matrix_multiplier.c
#include <stdlib.h>
#include "matrix_multiplier.h"

void initialize_matrices(int n, double a[n][n], double b[n][n]) {
    for (int i = 0; i < n; ++i) {
        for (int j = 0; j < n; ++j) {
            a[i][j] = (double) rand() / RAND_MAX;
            b[i][j] = (double) rand() / RAND_MAX;
        }
    }
}

void multiply(int n, double a[n][n], double b[n][n], double c[n][n]) {
    for (int i = 0; i < n; ++i) {
        for (int j = 0; j < n; ++j) {
            c[i][j] = 0;
            for (int k = 0; k < n; ++k) {
                c[i][j] += a[i][k] * b[k][j];
            }
        }
    }
}

File: Individual Assingment/TASK1/code/py/matrix_multiplier.py
import random

def initialize_matrices(n):
    A = [[random.random() 
          for _ in range(n)] 
          for _ in range(n)]
    B = [[random.random() 
          for _ in range(n)] 
          for _ in range(n)]
    C = [[0 
          for _ in range(n)] 
          for _ in range(n)]
    return A, B, C

def multiply(n, A, B, C):
    for i in range(n):
        for j in range(n):
            C[i][j] = 0
            for k in range(n):
                C[i][j] += A[i][k] * B[k][j]